### PR TITLE
grpc-js: stricter function check than instanceof

### PR DESCRIPTION
**Pull request by zereraz**
`instanceof Function` does not work in vm context, related to https://github.com/grpc/grpc-node/pull/204 https://github.com/grpc/grpc-node/pull/795 and https://github.com/grpc/grpc-node/pull/1759

### CODE REVIEW

File: packages/grpc-js/src/client.ts [hunk 1]
```diff
@@ -198,9 +198,9 @@ export class Client {
     options: CallOptions;
     callback: UnaryCallback<ResponseType>;
   } {
-    if (arg1 instanceof Function) {
+    if (Object.prototype.toString.call(arg1) === '[object Function]') {
```
Comment (murgatroid99) [inline packages/grpc-js/src/client.ts:201]: Can you extract this condition into an `isFunction` check?

Comment (zereraz) [reply packages/grpc-js/src/client.ts:201]: certainly, in this file itself, outside the class?

**Comment by murgatroid99** [comment]
It looks like this doesn't compile. The test log is here: https://source.cloud.google.com/results/invocations/f8f4ecea-d9b2-4d70-8893-604b5a7f30b0/targets/grpc%2Fnode%2Fpull_request%2Flinux/log. Can you fix that?

**Review by murgatroid99 — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**